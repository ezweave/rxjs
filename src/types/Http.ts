import { Observable } from 'rxjs';
import { HttpRequestConfig, HttpResponse } from 'rxjs-http-client';

/**
 * Separate because config is optional
 */
export interface GetHttpRequestFunction {
  (url: string, config?: Partial<HttpRequestConfig>): Observable<HttpResponse>;
}

interface BaseHttpRequestFunction {
  (url: string, config: Partial<HttpRequestConfig>): Observable<HttpResponse>;
}

export type DeleteHttpRequestFunction = BaseHttpRequestFunction;
export type PatchHttpRequestFunction = BaseHttpRequestFunction;
export type PostHttpRequestFunction = BaseHttpRequestFunction;
export type PutHttpRequestFunction = BaseHttpRequestFunction;