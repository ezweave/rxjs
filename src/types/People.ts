export interface People {
  birthYear: string;
  created: Date | string;
  edited: Date | string;
  eyeColor: string;
  films: string[];
  gender: string;
  hairColor: string;
  height: string;
  homeworld: string;
  mass: string;
  name: string;
  skinColor: string;
  species: any[];
  starships: any[];
  url: string;
  vehicles: any[];
}
