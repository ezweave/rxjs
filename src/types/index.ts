export * from './GetCraftAndCrew';
export * from './People';
export * from './Starship';
