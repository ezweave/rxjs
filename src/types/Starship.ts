import { People } from './People';

export interface Starship {
  cargoCapacity: string;
  consumables: string;
  costInCredits: string;
  created: Date | string;
  crew: string;
  edited: Date | string;
  films: string[];
  hyperdriveRating: string;
  length: string;
  manufacturer: string;
  maxAtmospheringSpeed: string;
  mglt: string;
  model: string;
  name: string;
  passengers: string | People[];
  pilots: string[];
  starshipClass: string;
  url: string;
}
