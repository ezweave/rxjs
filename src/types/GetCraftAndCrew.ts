import { Axios } from 'axios';

import { Starship } from './Starship';

export interface GetCraftAndCrew {
  (n: number, httpGet?: Axios['get']): Promise<Starship[][]>;
}
