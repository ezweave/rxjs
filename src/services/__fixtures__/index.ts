import { Starship } from '@rx-example/types';
import ImperialShuttleJSON from './ImperialShuttle.json';
import XWingJSON from './XWing.json';
import { transformAllFieldNamesToCamel } from 'field-caser';

export * from './LukeSkywalker';

export { ImperialShuttleJSON, XWingJSON };

export const ImperialShuttle = <Starship>(
  transformAllFieldNamesToCamel(ImperialShuttleJSON)
);
export const XWing = <Starship>transformAllFieldNamesToCamel(XWingJSON);
