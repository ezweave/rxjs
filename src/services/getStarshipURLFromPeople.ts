import { get } from 'lodash';

import { People } from '@rx-example/types';

export const getStarshipURLFromPeople = (people: People): string[] =>
  get(people, 'starships', ['https://swapi.dev/api/starships/11/']);
