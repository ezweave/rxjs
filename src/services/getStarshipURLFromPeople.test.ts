import { People } from '@rx-example/types';
import { LukeSkywalkerPeople } from './__fixtures__';
import { getStarshipURLFromPeople } from './getStarshipURLFromPeople';

describe(getStarshipURLFromPeople, () => {
  it('returns an array of starships, if found', () => {
    expect(getStarshipURLFromPeople(LukeSkywalkerPeople)).toEqual(
      LukeSkywalkerPeople.starships,
    );
  });
  it('returns a y-wing if you have no starships', () => {
    const noStarships = {} as unknown as People;
    expect(getStarshipURLFromPeople(noStarships)).toMatchSnapshot();
  });
});
