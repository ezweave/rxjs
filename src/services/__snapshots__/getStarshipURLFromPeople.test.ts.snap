// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getStarshipURLFromPeople returns a y-wing if you have no starships 1`] = `
Array [
  "https://swapi.dev/api/starships/11/",
]
`;
